src/box/txn.c

```diff
@@ -637,6 +638,20 @@ txn_on_journal_write(struct journal_entry *entry)
 	 */
 	assert(in_txn() == NULL);
 	fiber_set_txn(fiber(), txn);
+	/*
+	 * Use session and credentials of the original fiber for
+	 * commit/rollback triggers.
+	 */
+	struct session *orig_session = fiber_get_session(fiber());
+	struct session *session = (txn->fiber != NULL ?
```

Comment:
NIt: I would simply write
```
if (txn->fiber != NULL) {
        fiber_set_session(fiber(), fiber_get_session(txn->fiber));
        fiber_set_user(fiber(), fiber_get_user(txn->fiber));
}
```

Reply:
A fiber may have no session/user (`fiber_get_session()` and `fiber_get_user()` may return `NULL`). In this case we shouldn't set the current fiber's session/user to `NULL`, because it could result in creating a new session on demand (see `current_session()`), which would be wrong, because the session would be leaked when we restore the original session/user.